node/config/load.go

```diff
@@ -46,7 +70,70 @@ func FromReader(reader io.Reader, def interface{}) (interface{}, error) {
 	return cfg, nil
 }
 
-func ConfigUpdate(cfgCur, cfgDef interface{}, comment bool) ([]byte, error) {
+type cfgLoadOpts struct {
+	canFallbackOnDefault func() error
+	validate             func(string) error
+}
+
+type LoadCfgOpt func(opts *cfgLoadOpts) error
+
+func SetCanFallbackOnDefault(f func() error) LoadCfgOpt {
+	return func(opts *cfgLoadOpts) error {
+		opts.canFallbackOnDefault = f
+		return nil
+	}
+}
+
+func SetValidate(f func(string) error) LoadCfgOpt {
+	return func(opts *cfgLoadOpts) error {
+		opts.validate = f
+		return nil
+	}
+}
+
+func NoDefaultForSplitstoreTransition() error {
+	return xerrors.Errorf("FullNode config not found and fallback to default disallowed while we transition to splitstore discard default.  Use `lotus config default` to set this repo up with a default config.  Be sure to set `EnableSplitstore` to `false` if you are running a full archive node")
+}
+
+func ValidateSplitstoreSet(cfgRaw string) error {
+	explicitSplitstoreRx := regexp.MustCompile("\n[\t\n\f\r ]*EnableSplitstore =")
```

Comment:
`"(?m)^\s*EnableSplitstore\s*="`

- multiline (`^` matches beginning of line)
- match any number of non-line whitespace
- the string
- non-line whitespace
- equals